robozonky-strategy-natural/src/main/java/com/github/robozonky/strategy/natural/NaturalLanguageInvestmentStrategy.java

```diff
@@ -72,7 +73,11 @@ private boolean needsConfirmation(final LoanDescriptor loanDescriptor) {
                 .peek(d -> LOGGER.trace("Evaluating {}.", d.item()))
                 .map(l -> { // recommend amount to invest per strategy
                     final int recommendedAmount = recommender.apply(l.item(), balance);
-                    return l.recommend(recommendedAmount, needsConfirmation(l));
+                    if (recommendedAmount > 0) {
+                        return l.recommend(recommendedAmount, needsConfirmation(l));
+                    } else {
+                        return Optional.<RecommendedLoan>empty();
+                    }
```

Comment:
@jhrcek FYI This fixes the WARN you saw in your log. This doesn't change the functionality a bit, as the empty `Optional` was returned anyway. But this time, it will be returned without the warning. Good catch!